client/utils/custom-hooks.js

```diff
@@ -13,3 +13,39 @@ export const useDidUpdate = (callback, deps) => {
     }
   }, deps);
 };
+
+// Usage: const ref = useModalBehavior(() => setSomeState(false))
+// place this ref on a component
+export const useModalBehavior = (hideOverlay) => {
+  const ref = useRef({});
+
+  // Return values
+  const setRef = (r) => { ref.current = r; };
+  const [visible, setVisible] = useState(true);
+  const trigger = () => setVisible(true);
+
+  const hide = () => setVisible(false);
+
+
+  const handleClickOutside = ({ target }) => {
+    if (ref && ref.current && !ref.current.contains(target)) {
+      hide();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => document.removeEventListener('mousedown', handleClickOutside);
+  }, [ref]);
+
+  return [visible, trigger, setRef];
+};
+
+// TODO: This is HOC, not a hook. Where do we put it?
```

Comment:
I think HoCs can be placed in the `components` folder? That's what I've done for the other ones.